lib/kafka_ex/protocol/sync_group.ex

```diff
@@ -0,0 +1,76 @@
+defmodule KafkaEx.Protocol.SyncGroup do
+  import KafkaEx.Protocol.Common
+
+  @member_assignment_version 0
+
+  defmodule Assignment do
+    defstruct topic: nil, partitions: []
+    @type t :: %Assignment{topic: binary, partitions: [integer]}
+  end
+
+  defmodule Response do
+    defstruct error_code: nil, assignments: []
+    @type t :: %Response{error_code: atom | integer, assignments: [Assignment.t] }
+  end
+
+  @spec create_request(integer, binary, binary, integer, binary, [{binary, [Assignment.t]}]) :: binary
+  def create_request(correlation_id, client_id, group_name, generation_id, member_id, assignments) do
+    KafkaEx.Protocol.create_request(:sync_group, correlation_id, client_id) <>
+    << byte_size(group_name) :: 16-signed, group_name :: binary,
+       generation_id :: 32-signed,
+       byte_size(member_id) :: 16-signed, member_id :: binary,
+       length(assignments) :: 32-signed, group_assignment_data(assignments, "") :: binary,
+       0 :: 32-signed
+    >>
+  end
+
+  @spec parse_response(binary) :: Response.t
+  def parse_response(<< _correlation_id :: 32-signed, error_code :: 16-signed,
+                        _member_assignment_len :: 32-signed,
```

Comment:
Something doesn't line up here.  The guide says the format of the response is

```
SyncGroupResponse => ErrorCode MemberAssignment
  ErrorCode => int16
  MemberAssignment => bytes
```

And, afaict, MemberAssignment is

```
MemberAssignment => Version PartitionAssignment
  Version => int16
  PartitionAssignment => [Topic [Partition]]
    Topic => string
    Partition => int32
  UserData => bytes
```

So I'm not seeing where `_member_assign_len :: 32-signed` fits in.

Also, what is `version` here?  Is it a protocol version or a version of the group structure?  The protocol is pegged at 0, but the version of the group structure will not be.  I can try to dig through the docs more tomorrow for details.